controller_manager/src/controller_manager.cpp

```diff
@@ -230,6 +233,39 @@ ControllerManager::ControllerManager(
     "Controllers Activity", this, &ControllerManager::controller_activity_diagnostic_callback);
 }
 
+ControllerManager::ControllerManager(
+  std::shared_ptr<rclcpp::Executor> executor, const std::string & urdf,
+  bool activate_all_hw_components, const std::string & manager_node_name,
+  const std::string & node_namespace, const rclcpp::NodeOptions & options)
+: rclcpp::Node(manager_node_name, node_namespace, options),
+  update_rate_(get_parameter_or<int>("update_rate", 100)),
+  resource_manager_(std::make_unique<hardware_interface::ResourceManager>(
+    urdf, this->get_node_clock_interface(), this->get_node_logging_interface(),
+    activate_all_hw_components, update_rate_)),
+  diagnostics_updater_(this),
+  executor_(executor),
+  loader_(std::make_shared<pluginlib::ClassLoader<controller_interface::ControllerInterface>>(
+    kControllerInterfaceNamespace, kControllerInterfaceClassName)),
+  chainable_loader_(
+    std::make_shared<pluginlib::ClassLoader<controller_interface::ChainableControllerInterface>>(
+      kControllerInterfaceNamespace, kChainableControllerInterfaceClassName))
+{
+  if (!get_parameter("update_rate", update_rate_))
+  {
+    RCLCPP_WARN(get_logger(), "'update_rate' parameter not set, using default value.");
+  }
+
+  if (is_resource_manager_initialized())
+  {
+    init_services();
+  }
+  subscribe_to_robot_description_topic();
```

Comment:
Yes, we have it in other constructors too. I would leave it here, in case the user passes empty string or the URDF is invalid and the resource manager is not properly initialized and when he publishes a new URDF the ros2_control will automatically configure the ResourceManager